admin/tests/integration/components/campaigns/update_test.js

```diff
@@ -0,0 +1,129 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { click, fillIn, render } from '@ember/test-helpers';
+import sinon from 'sinon';
+import hbs from 'htmlbars-inline-precompile';
+import EmberObject from '@ember/object';
+
+module('Integration | Component | Campaigns | Update', function(hooks) {
+  setupRenderingTest(hooks);
+
+  let toggleEditMode;
+  let update;
+  let campaign;
+
+  hooks.beforeEach(function() {
+    update = sinon.stub();
+    toggleEditMode = sinon.stub();
+    const save = sinon.stub();
+    campaign = EmberObject.create({ title: 'Ceci est un titre', name: 'Ceci est un nom', save });
+
+    this.set('update', update);
+    this.set('toggleEditMode', toggleEditMode);
+    this.set('campaign', campaign);
+  });
+
+  test('it should display the items', async function(assert) {
+    // when
+    await render(hbs`<Campaigns::update @campaign={{this.campaign}} @toggleEditMode={{this.toggleEditMode}} />`);
+    // then
+    assert.equal(this.element.querySelector('label[for="name"]').textContent.trim(), 'Nom de la campagne');
+    assert.equal(this.element.querySelector('label[for="customLandingPageText"]').textContent.trim(), 'Texte de la page d\'accueil');
+    assert.equal(this.element.querySelector('#name').value, 'Ceci est un nom');
+    assert.contains('Annuler');
+    assert.contains('Enregistrer');
+  });
+
+  module('when campaign is of type assessment', function(hooks) {
+    hooks.beforeEach(() => {
+      campaign.isTypeAssessment = true;
+    });
+
+    test('it should display items for assessment', async function(assert) {
+      // when
+      await render(hbs`<Campaigns::update @campaign={{this.campaign}} @toggleEditMode={{this.toggleEditMode}} />`);
+      // then
+      assert.equal(this.element.querySelector('label[for="title"]').textContent.trim(), 'Titre du parcours');
+      assert.equal(this.element.querySelector('label[for="customResultPageText"]').textContent.trim(), 'Texte de la page de fin de parcours');
+      assert.equal(this.element.querySelector('label[for="customResultPageButtonText"]').textContent.trim(), 'Texte du bouton de la page de fin de parcours');
+      assert.equal(this.element.querySelector('label[for="customResultPageButtonUrl"]').textContent.trim(), 'URL du bouton de la page de fin de parcours');
+      assert.equal(this.element.querySelector('#title').value, 'Ceci est un titre');
```

Comment:
[suggestion]
Au lieu d'utiliser `assert.equal(this.element.querySelector('label[for="title"]'))`, on peut utiliser `assert.dom('label[for="title"]').hasText()` ou `assert.dom('label[for="title"]').includesText()`

Et encore mieux serait d'avoir une fonction utilitaire `findByLabel` (qu'on pourrait extraire des fonctions utilitaires existantes `fillInByLabel` et `clickByLabel`) et on pourrait écrire:
```
assert.ok(findByLabel('Titre du parcours'));
```

Reply:
mais dans un premier temps je vais utiliser assert.dom